### PR TITLE
playwright added, landing screen test added, Vitest added and unit test added for  landing screen.

### PR DESCRIPTION
**Pull request by jahanvi-baragaun**
## What does this PR do?
added [Playwright](https://playwright.dev/),  an end-to-end testing framework.
added Vitest, unit test framework.

### Related Issues & PRs
Fixes https://github.com/baragaun/first-spark-app/issues/22

#### Key Decisions
- **Implemeneted Approach**:
  [Playwright](https://playwright.dev/)
  Vitest
- **Alternatives Considered**:
  [Cypress](https://www.cypress.io/)
- **Rationale**:
  Research says:

Browser Support                       =    Playwright
Speed & Performance               =   Playwright
Debugging & UI                           =  Cypress
Multiple Tabs & Mobile Testing   =  Playwright
Ease of Use.                                   =  Cypress
Parallel Testing (Free)                    = Playwright

## Breaking Changes
- [x] This PR includes breaking changes
   To clean project
        rm -rf node_modules package-lock.json
        pnpm i
  Please run following command  to run the tests:
    To run unit tests:   pnpm run test
    To run end-to-end integration tests (any from below commands):   
           1.  pnpm playwright test --headed 
           2. pnpm playwright test
           3. pnpm playwright test: e2e
    To run all tests:  pnpm run test:all

### Limitations & Concerns
- **Known limitations**: NA
- **Edge cases**: NA
- **Potential risks**: NA

---

### UI Changes
- [ ] This PR includes UI changes

### Screenshots
No UI changes

---

### Reviewers

- [ ] Reviewer 1: @RaghvindYadav 
- [ ] Reviewer 2: @jonahtwalker 

### Next Steps

- [ ] Follow-up tasks are created
  - [ ] setup vitest for unit testing
  - [ ] setup workflow so that integration tests run on PR merge/create

### How to Test

1. Prerequisites: NA
5. Test steps:
        rm -rf node_modules package-lock.json
        pnpm install
    To run unit tests:   pnpm run test
    To run end-to-end integration tests (any from below commands):   
           1.  pnpm playwright test --headed 
           2. pnpm playwright test
           3. pnpm playwright test: e2e
    To run all tests:  pnpm run test:all
        
6. Expected results:
       Landing screen  UI tests should pass
       Nav bar unit test should pass 
       

## Final Checklist
- [ ] Linked related issues
- [ ] Rebased with target branch
- [ ] No lint errors
- [ ] QA'd in multiple browsers
- [ ] Added/updated tests (if applicable)
- [ ] Mobile responsive (if applicable)
- [ ] Updated documentation (if applicable)
- [ ] Notified the Team of this updates readiness

---

## AI Usage

- [x] AI tools were used in development
  - **Tool used**:
    ChatGPT
    Augment Code
  - **Prompts used**:
   Comparison between different e2e frameworks for Svelte5 project - ChatGPT
   Please add test in landing.spac.ts regarding navElement contains "Sign Up" button.
   create a git workflow file to run integration tests 
    - **Initial prompt**: 
- [ ] No AI tools were used
  - **Reason**:

## References

- **Documentation**:
- **Discussions**:
- **Related PRs**:

### CODE REVIEW

**Comment by RaghvindYadav** [comment]
Thank you Jahanvi, I checked the PR. It's look good.

**Comment by RaghvindYadav** [comment]
> To run end-to-end integration tests (any from below commands):
> 1. pnpm playwright test --headed
> 2. pnpm playwright test
> 3. pnpm playwright test: e2e
> To run all tests: pnpm run test:all

Make sure the terminal code is corresponding to `scripts` set which is available in package.json

File: package.json [hunk 1]
```diff
@@ -12,21 +12,29 @@
...
+    "test:all": "vitest run && playwright test"
   },
   "devDependencies": {
     "@eslint/compat": "^1.2.5",
     "@eslint/js": "^9.18.0",
+    "@playwright/test": "^1.50.1",
     "@sveltejs/adapter-auto": "^4.0.0",
     "@sveltejs/kit": "^2.16.0",
     "@sveltejs/vite-plugin-svelte": "^5.0.0",
+    "@testing-library/jest-dom": "^6.6.3",
```
Comment (shadowbrush) [inline package.json:28]: I now understand why you pulled in a library from Jest: https://markus.oberlehner.net/blog/using-testing-library-jest-dom-with-vitest/

**Review by shadowbrush — APPROVED** [review]
Looks good! Excited to have tests in the app, and e2e no less! 👍

**Comment by shadowbrush** [comment]
@jahanvi-baragaun Before merging this, can you please add the instructions on how to run the tests to the `README.md`? 

Also, this gave me an error: `pnpm playwright test: e2e`.

**Comment by jahanvi-baragaun** [comment]
> @jahanvi-baragaun Before merging this, can you please add the instructions on how to run the tests to the `README.md`?
> 
> Also, this gave me an error: `pnpm playwright test: e2e`.

Sure. I will add in Readme.md.  Thanks.

File: docs/svelte-complete-medium.txt
Comment (jahanvi-baragaun) [inline docs/svelte-complete-medium.txt:1]: @jonahtwalker  I have added this file to help AI to understand Svelte5 syntax. 
reference: https://khromov.se/getting-better-ai-llm-assistance-for-svelte-5-and-sveltekit/?utm_source=chatgpt.com

**Comment by jahanvi-baragaun** [comment]
@jonahtwalker   I m getting below error for the test workflow.\
ERROR  Command failed with exit code 128: git ls-remote git+ssh://git@github.com/baragaun/bg-node-client.git HEAD
git@github.com: Permission denied (publickey).

is there any token used I secrets which I can use?  like  token: ${{ secrets.GITHUB_TOKEN }} in workflow?

File: docs/svelte-complete-medium.txt
Comment (jonahtwalker) [inline docs/svelte-complete-medium.txt:1]: 👍🏼

**Review by jonahtwalker — APPROVED** [review]
Thank you Jahanvi, this is great to have from the start. 🚀 Getting the latest from `milestone-01` should fix your node-client issues. 👍🏼

**Comment by jonahtwalker** [comment]
@jahanvi-baragaun Are you going to be adding more tests or are you anticipating that this is merged before that?

I think a rebase would love the .lock conflicts here

**Comment by jonahtwalker** [comment]
There is an issue for this: https://github.com/baragaun/first-spark-app/issues/22

Please include these in the description instead of `N/A`

**Comment by jahanvi-baragaun** [comment]
> There is an issue for this: #22
> 
> Please include these in the description instead of `N/A`

oh ok. got it. Thank you.

**Comment by jahanvi-baragaun** [comment]
> @jahanvi-baragaun Are you going to be adding more tests or are you anticipating that this is merged before that?
> 
> I think a rebase would love the .lock conflicts here

I will not be adding more tests for now. will rebase to resolve conflicts.
